src/httpclient.c

```diff
@@ -217,11 +243,51 @@ httpclient_Result httpclient_post(
 static void httpclient_tlsConfigFree(void *o) {
     httpclient_Config config = (httpclient_Config)o;
     if (config) {
+        if (config->user) {
+            corto_dealloc(config->user);
+        }
+
+        if (config->password) {
+            corto_dealloc(config->password);
+        }
+
         free(config);
     }
 
 }
 
+httpclient_Config httpclient_Config__create(void) {
+    httpclient_Config o = (httpclient_Config)malloc(
+        sizeof(struct httpclient_Config_s));
+
+    o->user = NULL;
+    o->password = NULL;
+    o->timeout = DEFAULT_TIMEOUT;
+    o->connectTimeout = DEFAULT_CONNECT_TIMEOUT;
+
+    if (!o) {
+        corto_throw("Failed to initialize configuration data");
```

Comment:
This condition is redundant. You were already able to dereference `o` when you get to this point, so it can't be NULL. If you want to check whether malloc returns `NULL`, you need to move up this statement.